charms/slurmctld/src/interface_slurmd.py

```diff
@@ -124,7 +141,10 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
                     if node_config := node.get("node_parameters"):
                         if node_name := node_config.get("NodeName"):
                             self._charm.new_nodes = list(set(self._charm.new_nodes + [node_name]))
-                            self.on.slurmd_available.emit()
+                            self.on.slurmd_available.emit(
+                                node_name=node_name, gres_info=node.get("gres")
+                            )
+                            logger.debug(f"_on_relation_changed node_config = {node_config}")
```

Comment:
This is something we need to do on most of our repos, but we should be using the logger's built-in formatter rather than f-strings. It isn't as readable unfortunately, [but it is more optimized](https://docs.python.org/3/howto/logging.html#optimization). The logger's built-in formatter will lazily format the message when absolutely necessary while f-strings will immediately format the message which isn't always optimal, especially if the value is costly to format:

```suggestion
                            logger.debug(f"_on_relation_changed node_config = %s", node_config)
```

_Let's get in the practice of using the built-in formatter now rather than later so refactoring it later won't be as painful_ :sweat_smile:

Reply:
You can just make the modification to the logging format in your current changes, not the entire charm. We will save that for when we're close to the end of this cycle :smile:

Reply:
Done.